staging/com.smartnews/manifest.tt

```diff
@@ -0,0 +1,37 @@
+class @com.smartnews
+{
+  import loader from @org.thingpedia.v2();
+  import config from @org.thingpedia.config.none();
+
+  list query top_articles(in opt counter: Number #[prompt="How many news articles do you want?"],
+    			  out news_id: String,
```

Comment:
This parameter should be called `id` and should be of Entity type. You should declare the entity type with a statement in the class:
```
entity article #_[name="Articles on SmartNews"] #[has_ner=true];
```

The `#[has_ner]` annotation can be true or false, depending on whether the "title" of the article should be the "name" of the article.
If you set it to `true`, you should remove the `title` parameter, and in your code return `id : new Tp.Value.Entity(news_id, title)`. In that case, you get decent reply templates of the form: "I found <title>, it's an article by <author> etc. etc."
If you set it to `false`, the "id" will be consider an _unnamed entity_, and you won't get those default templates for replies. I'm not sure exactly what kind of replies we offer in that case, we probably need to tweak the state machine a bit.

Reply:
The id is for internal reference to the specific news article here. What would be the recommended way to implement it? The old implementation is to let users submit the id so they can save it to the reading list (which is not very convenient), for example: pocket smartnews "4136423963234175060".

Reply:
I don't expect we'll care to support `pocket smartnews "4136423963234175060"` because nobody knows those IDs. You will get `pocket this news article` as a valid command if the last command returned a news article (something with `id` of the right type).

All things considered, I think ID should be a named entity, with the internal ID number as the `value` of the entity, and the article title as the `display` of the entity.